### PR TITLE
🔥 Ze ultimate PR 🔥

### PR DESCRIPTION
**Pull request by 3nids**
☣

### CODE REVIEW

**Comment by 3nids** [comment]
⛵

image

**Comment by 3nids** [comment]
v2, still not perfect
image

**Comment by 3nids** [comment]
If ok, you can merge https://github.com/qgis/QGIS/pull/8085

File: src/app/qgsgeometryvalidationdock.cpp [hunk 1]
```diff
@@ -182,6 +204,27 @@ void QgsGeometryValidationDock::onCurrentErrorChanged( const QModelIndex &curren
...
+void QgsGeometryValidationDock::onCurrentLayerChanged( QgsMapLayer *layer )
+{
+  // activate icon
+  bool enabled = false;
+  QgsVectorLayer *vl = dynamic_cast<QgsVectorLayer *>( layer );
+  if ( vl && vl->isSpatial() )
+  {
+    const QList<QgsGeometryCheckFactory *> topologyCheckFactories = QgsAnalysis::instance()->geometryCheckRegistry()->geometryCheckFactories( vl, QgsGeometryCheck::LayerCheck, QgsGeometryCheck::Flag::AvailableInValidation );
+    const QStringList activeChecks = vl->geometryOptions()->geometryChecks();
+    for ( const QgsGeometryCheckFactory *factory : topologyCheckFactories )
```
Comment (m-kuhn) [inline src/app/qgsgeometryvalidationdock.cpp]: Let's send this through the geometry validation service which already keeps track of that.

**Pull request merged**